kmk/ble.py

```diff
@@ -0,0 +1,64 @@
+from adafruit_ble import BLERadio
+from adafruit_ble.advertising import Advertisement
+from adafruit_ble.advertising.standard import ProvideServicesAdvertisement
+from adafruit_ble.services.standard.hid import HIDService
+from kmk.hid import (
+    HID_REPORT_SIZES,
+    AbstractHID,
+    HIDReportTypes,
+    HIDUsage,
+    HIDUsagePage,
+)
+
+BLE_APPEARANCE_HID_KEYBOARD = 961
+
+
+class BLEHID(AbstractHID):
+    def post_init(self):
+        self.devices = {}
+
+        hid = HIDService()
+
+        advertisement = ProvideServicesAdvertisement(hid)
+        advertisement.appearance = BLE_APPEARANCE_HID_KEYBOARD
+        scan_response = Advertisement()
+        scan_response.complete_name = 'KMK Keyboard'
+
+        ble = BLERadio()
+        if not ble.connected:
+            ble.start_advertising(advertisement, scan_response)
+            while not ble.connected:
+                pass
```

Comment:
This is *really* aggressive - it means we'll theoretically "snap out of it" faster when a connection is established, but in the mean time will draw more power.

This is okay for now - I eventually want to do a second pass on BLE support some day anyway.

Reply:
This is a pretty standard behavior on keyboards as it's meant to "snap out of it" fast. We may be able to do some sort of software switch to "turn off" though most keyboards have a physical switch to the battery. If we have a key to switch between "bluetooth mode" and "hid mode" this could also force a bypass as a "sort of off" state none the less.